sale_delivery_state/models/sale_order.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Pierrick BRUN <pierrick.brun@akretion.com>
+# Copyright 2018 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    delivery_state = fields.Selection(
+        [
+            ("no", "No delivery"),
+            ("unprocessed", "Unprocessed"),
+            ("partially", "Partially processed"),
+            ("done", "Done"),
+        ],
+        string="Delivery state",
+        # Compute method have a different name then the field because
+        # the method _compute_delivery_state already exist to compute
+        # the field delivery_set
+        compute="_compute_sale_delivery_state",
+        store=True,
+    )
+
+    force_delivery_state = fields.Boolean(
+        string="Force delivery state",
+        help=(
+            "Allow to enforce done state of delivery, for instance if some"
+            " quantities were cancelled"
+        ),
+    )
+
+    def _all_qty_delivered(self):
+        """
+        Returns True if all line have qty_delivered >= to ordered quantities
+
+        If `delivery` module is installed, ignores the lines with delivery costs
+
+        :returns: boolean
+        """
+        self.ensure_one()
+        # Skip delivery costs lines
+        sale_lines = self.order_line.filtered(lambda rec: not rec._is_delivery())
```

Comment:
@PierrickBrun wondering: does it make any sense to rely on `_is_delivery` only?
IMO we should depend on `delivery` and then use `is_delivery` flag instead (which is what is used by `_is_delivery` when `delivery` module is installed. 
I mean, what's the point of having a delivery state that won't be ever a delivery if you don't install `delivery`? :thinking: 
This would also make testing easier (no mocking required).

Reply:
I think the module was named something else when this decision was made but it may be pointless now.

Since this module is becoming a dependecy of sale_automatic_workflow, maybe it is a good idea to keep it that way to avoid having `sale_automatic_workflow` depend on `delivery` ?

Reply:
good point! :+1: 

Can you address this https://github.com/OCA/sale-workflow/pull/1372#discussion_r665419433 and cherry-pick my tests refactoring? :pray:

Reply:
Done, thank you for the tests refactoring